src/pages/challenges.tsx

```diff
@@ -30,25 +36,27 @@ const Challenges: FC = () => {
   }, []);
 
   useEffect(() => {
-    if ('challenges' in response) {
-      let challenges: challenge['content'] = response.challenges;
-      const challengeMap: any = [];
-
-      challengeMap.Active = challenges.filter((task) => task.is_active);
-      challengeMap.Completed = challenges.filter((task) => !task.is_active);
-
-      setFilteredChallenge(challengeMap);
+    if (isLoggedIn && !Object.keys(response).length) {
+      callAPI();
+      if ('challenges' in response) {
+        let challenges: challenge['content'] = response.challenges;
+        const challengeMap: any = [];
```

Comment:
Isn't `challengeMap` should be `{}` object?

Reply:
@Maheima could you create issues for fixing this? Moving forward it will create confusion.